addons/crm/static/src/js/forecast/forecast_search_model.js

```diff
@@ -0,0 +1,106 @@
+/** @odoo-module */
+
+import { Domain } from "@web/core/domain";
+import { makeContext } from "@web/core/context";
+import { SearchModel } from "@web/search/search_model";
+
+/**
+ * This is the conversion of ForecastModelExtension. See there for more
+ * explanations of what is done here.
+ */
+const DATE_FORMAT = {
+    datetime: "YYYY-MM-DD HH:mm:ss",
+    date: "YYYY-MM-DD",
+};
+
+export class ForecastSearchModel extends SearchModel {
+    /**
+     * @override
+     */
+    exportState() {
+        const state = super.exportState();
+        state.forecast = {
+            forecastField: this.forecastField,
+            forecastFilter: this.forecastFilter,
+            forecastStart: this.forecastStart,
+        };
+        return state;
+    }
+
+    /**
+     * @protected
+     * @returns {string}
+     */
+    _getForecastStart() {
+        /** @todo stop using moment */
+        const type = this.searchViewFields[this.forecastField].type;
+        let startMoment;
+        if (type === "date") {
+            const groupBy = this.groupBy;
+            const firstForecastGroupBy = groupBy.find((gb) => gb.includes(this.forecastField));
+            let granularity = "month";
+            if (firstForecastGroupBy) {
+                granularity = firstForecastGroupBy.split(":")[1] || "month";
+            } else if (groupBy.length) {
+                granularity = "day";
+            }
+            startMoment = moment().startOf(granularity);
+        } else {
+            startMoment = moment.utc(startMoment);
+        }
```

Comment:
```javascript
const groupBy = this.groupBy;
const firstForecastGroupBy = groupBy.find((gb) => gb.includes(this.forecastField));
let granularity = "month";
if (firstForecastGroupBy) {
    granularity = firstForecastGroupBy.split(":")[1] || "month";
} else if (groupBy.length) {
    granularity = "day";
}
startMoment = moment().startOf(granularity);
if (type === "datetime") {
    startMoment = moment.utc(startMoment);
}
```

Reply:
thx

Reply:
An idea: I think it would have been simpler to have <filter name="..." context="{ 'forecast_field': 'my_date_field' }"/> (and not put smth in the action context) and have a state containing only the forecastStart. If you agree it would be nice to migrate to such a system at some point.